### PR TITLE
refactor: use handlebars for views

### PR DESCRIPTION
**Pull request by JustinBeckwith**
First draft.  I noticed there was a fair bit of hand rolled view rendering happening, so I went ahead and moved us towards static serving for assets, and a view template rendering system with handlebars. 

Also lifts the JavaScript out of the `html` files, and sticks them in their own `js` files.

### CODE REVIEW

File: public/manage-tokens.js [hunk 1]
```diff
@@ -0,0 +1,268 @@
+/**
```
Comment (JustinBeckwith) [inline public/manage-tokens.js]: This code was just moved from `manage-tokens.html`

File: public/token-settings.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
+/**
```
Comment (JustinBeckwith) [inline public/token-settings.js]: This code was just moved from `token-settings.html`

File: src/server.ts [hunk 3]
```diff
@@ -15,65 +15,42 @@
...
+    defaultLayout: 'main',
+    partialsDir: path.join(__dirname, '../../views/partials'),
+    layoutsDir: path.join(__dirname, '../../views/layouts'),
+  })
 );
-const favicon = readStatic('favicon.ico');
-const changelog = readStatic('changelog.html')
-  .toString('utf8')
-  // Reduce the heading level in the CHANGELOG by 1:
-  .replace(/h3/g, 'h4')
```
Comment (JustinBeckwith) [inline src/server.ts]: Just a heads up - this down-headerizing is missing from this solution.  Not sure how much we really cared about it.

File: src/server.ts [hunk 4]
```diff
@@ -109,21 +86,6 @@ app.use((req, res, next) => {
   next();
 });
 
-const serve = (p: string, data: string | Buffer, loginServer?: boolean) => {
-  app.get(p, (req, res) => {
-    if (loginServer && !config.loginEnabled) {
```
Comment (JustinBeckwith) [inline src/server.ts]: Calling this out - this code was protecting static assets (css, js, favicon, etc).  In the new code, it's unprotected.  I don't think this is a problem.

File: src/server.ts (hunk 3 above)
Comment (bcoe) [reply src/server.ts]: The CHANGELOG being embedded in the page starts to look a little overwhelming with these headers, and the fact that we have multiple `h1` tags breaks HTML best practices.

I think embedding the CHANGELOG is neat, but it does start to dominate that page, I wonder if there's a better approach, perhaps we should just link to it?

Comment (JustinBeckwith) [reply src/server.ts]: I'm happy to re-do the header downsizing, but agree that putting it on it's own page likely makes more sense.  You cool with linking out to it?

File: src/server.ts (hunk 4 above)
Comment (soldair) [reply src/server.ts]: thanks for calling this out. you need this now that javascript has been externalized form token settings.

we should have a system test of some sort in the future that makes sure all of our private routes are in fact private when express is really running.

Comment (JustinBeckwith) [reply src/server.ts]: Huh - are you saying that we need to protect the client side JavaScript?  On having system tests for private routes, yes, that seems like a great idea!

File: src/server.ts (hunk 3 above)
Comment (bcoe) [reply src/server.ts]: Yes let's just link to it.

File: src/server.ts (hunk 4 above)
Comment (soldair) [reply src/server.ts]: no. no need to protect client side js. =) just that we have routes that are securred based on the magic ordering of express binding and we should do better ;)

**Pull request merged**